### DISCUSSION TITLE
issue: The custom resolver function is firing onChange despite mode being onBlur

### DISCUSSION THREAD

**Discussion by made-in-nz** [react-hook-form/react-hook-form#9550] [Q&A] [answered]
### Version Number

7.40.0

### Codesandbox/Expo snack

https://codesandbox.io/s/5d9ics

### Steps to reproduce
(NOTE codesandbox has been updated with fix)
1. Go to https://codesandbox.io/s/5d9ics
2. **Note** - a codesandbox bug with bootstrap scsss means you'll need to hit the codesandbox browser refresh button to get the app to render.
3. Open the console
4. Enter data into Surname or Given names fields
5. Console logs from the customResolver function demonstrate it is being called each time the input changes.

### Expected behaviour

1. The resolver function should only be called when the input blurs.
2. The resolver function is also being called multiple times when the form first renders (sounds like the same as https://github.com/react-hook-form/react-hook-form/issues/3066)

The codesandbox above is a very stripped down version of an app I am working on.  You can ignore the Form1,Form2,Form3 components.  Essentially everything relevant should be in the Form4 and RHFInput components.  

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment, marked as answer]
The behaviour described is due to `formState.isValid` subscription inside the controlled component

**Comment by made-in-nz** [comment]
@Moshyfawn thank you!
That was my simple example.  In the same codesandbox I've now enabled three other tabs that demonstrate (with a stubbed custom resolver) how I'm managing a record made up of multiple forms.  Each form needs to be separate as it is validated against the server separately, and the AppRouter component needs to know if the form is valid so as to be able to display the correct state (colour) of the tab label.
The resolver is being called each time the a field on these forms change.  Even if I remove the formState.isValid subscriptions from AppRouter the resolver is still called on change.
So my questions are:
1. Wondering if there is a reason why react-hook-form needs to re-run the resolver in this situation?
2. On a related note - would it be possible for the resolver mode to be set to something like 'none' and *only* have the resolver called via trigger()?

**Comment by made-in-nz** [comment]
Ok so there were two issues in the codesandbox.  *Either* will cause the custom resolver to fire on input change.

1.  AppRouter subscribing to each form's formState.isValid.
2.  The @hookform/devtools DevTools component in use alongside FormProvider .  There is a note in the docs https://react-hook-form.com/advanced-usage#FormProviderPerformance about performance.

The codesandbox now has DevTools enabled on the first tab (Individual) and resolver is firing on change, but on remaining tabs without DevTools they do not fire on change.
